### ISSUE TITLE
FollowUpEndDate API does not deliver the expected results

### ISSUE THREAD

**Issue by blzofhk** [hzi-braunschweig/SORMAS-Project#4221]
## Issue Description
Hello, I am Zac from the PIA Team.

By testing the following API call: 
`https://sb.sormas.netzlink.com/sormas-rest/visits-external/followUpEndDates/0`

Only three entries are retrieved:

```
[ 
   { "personUuid": "QQBSF6-G3FVA5-CSLMIN-QZXC2M4Q", "latestFollowUpEndDate": 1603058400000 },
   { "personUuid": "RFKSMA-POBGTP-SXVNUY-A4I6SLHM", "latestFollowUpEndDate": 1597788000000 },
   { "personUuid": "VRS2OK-B52B5Z-FTP7FE-3T5RSEEE", "latestFollowUpEndDate": 1613084400000 } 
]
```
At least the following entry was additionally expected:
  ###
`"personUuid": "RRUZFH-TLU5YX-RYCGND-X2APSJJA", "latestFollowUpEndDate": null }` 

Moreover, if the Follow-Up was cancelled, the `latestFollowUpEndDate` must be returned as NULL

So, the issue we found consists of two problems:
- The API does not retrieve all expected results
- (not confirmed) The `latestFollowUpEndDate` is not retrieved as NULL if the Follow-Up was cancelled. 

## Related Feature
https://github.com/hzi-braunschweig/SORMAS-Project/issues/3294

**Comment by FredrikSchaeferVitagroup** [comment]
It should be sufficient to remove these two lines
```
filter = CriteriaBuilderHelper.and(cb, filter, cb.notEqual(contactRoot.get(Contact.FOLLOW_UP_STATUS), FollowUpStatus.CANCELED));
filter = CriteriaBuilderHelper.and(cb, filter, cb.notEqual(contactRoot.get(Contact.FOLLOW_UP_STATUS), FollowUpStatus.NO_FOLLOW_UP));
```
from the [PersonFacadeEjb.getContactLatestFollowUpEndDates(...)](https://github.com/hzi-braunschweig/SORMAS-Project/blob/0dcb04e0a425a41727985a73a65f5fe068453313/sormas-backend/src/main/java/de/symeda/sormas/backend/person/PersonFacadeEjb.java#L405).

**Comment by irinajanzen-hzi** [comment]
@FredrikSchaeferVitagroup did you fix the bug? I am still new to SORMAS processes and am not sure where I could look that up. thanks.

**Comment by FredrikSchaeferVitagroup** [comment]
Hello @irinajanzen-hzi. On the right side column, you find a project and a milestone section. That is where we usually update our status of the ticket. The milestone indicates in which sprint the ticket is supposed to be taken care of. The project section indicates which development team (application or interactions) is supposed to deal with it and in what state of work it is.
When a ticket is done, you usually find something like that in the project section: 
![image](https://user-images.githubusercontent.com/67001822/107353893-db98db80-6acd-11eb-86e5-22555aba00f0.png)
This tells you that the interactions team took care of it, and that it is done. As you can find nothing here, this ticket is not finished. It is not even assigned to a development team. @vidi42 could probably give this ticket a start.

**Comment by MateStrysewske** [comment]
@FredrikSchaeferVitagroup @ssferrazza Can you please investigate and open a new issue if necessary?

**Comment by ssferrazza** [comment]
@MateStrysewske Yes, I can check it

**Comment by FredrikSchaeferVitagroup** [comment]
@blzofhk There are several UUIDs in SORMAS, two of them are the Contact UUID and the Person UUID. PIA is a Person based system, that's why SORMAS always uses the Person UUID when communicating with PIA (as you can see in the responses you posted). **WJFVXC-EGEZPI-UQAHJQ-MLV3SNUY** is a Contact UUID, so you won't and shall never see it in PIA calls. 
Instead you need to look for the UUID of the related Person (which is **T6VCST-PJSBJI-UTHDVW-PKUESP2Q**).

The current problem is rather, that also for Contacts with cancelled follow-up, a follow-up end date is returned. This is to be fixed in this issue.

**Comment by RobYed** [comment]
@ssferrazza @FredrikSchaeferVitagroup 
Which SORMAS Version will the fix be part of?

**Comment by irinajanzen-hzi** [comment]
@ssferrazza @FredrikSchaeferVitagroup are the fixes in 1.57.1 oder 1.57.2 for this issue? Thanks!

**Comment by FredrikSchaeferVitagroup** [comment]
@irinajanzen-hzi We are still working on it. The fix will probably go into 1.58.0.
What is in a release or hotfix can usually be found out via the [release notes here](https://github.com/hzi-braunschweig/SORMAS-Project/releases) (there are not release notes for 1.57.2 yet, this comment is just meant to be a general hint).
